mzLib/Proteomics/ProteolyticDigestion/ProteolyticPeptide.cs

```diff
@@ -97,48 +97,51 @@ internal IEnumerable<PeptideWithSetModifications> GetModifiedPeptides(IEnumerabl
             }
 
             // LOCALIZED MODS
-            foreach (var kvp in Protein.OneBasedPossibleLocalizedModifications)
+            var localizedModsOrderedByPositionThenByModification = Protein.OneBasedPossibleLocalizedModifications
+                .SelectMany(kvp => kvp.Value.Select(value => (kvp.Key, value)))
+                .OrderBy(tuple => tuple.Key)
+                .ThenBy(tuple => tuple.value)
```

Comment:
Given how many times this gets called, I think it would be better to use a custom comparer, that way we only have to iterate through the list once.

Additionally, I'm not sure that it makes sense to store this as a list, since we're only iterating through it once. We could just put the linq expression in the foreach

Reply:
updated according to your comments